go-controller/pkg/ovn/egressip.go

```diff
@@ -376,14 +376,25 @@ func (oc *Controller) reconcileEgressIPPod(old, new *v1.Pod) (err error) {
 			// If the podSelector is not empty then check if the pod stopped
 			// matching. If the pod was deleted, "new" will be nil and
 			// newPodLabels will not match, so this is should cover that case.
-			if !podSelector.Empty() && !podSelector.Matches(newPodLabels) && podSelector.Matches(oldPodLabels) {
-				return oc.deletePodEgressIPAssignments(egressIP.Name, egressIP.Status.Items, oldPod)
+			if !podSelector.Empty() && !podSelector.Matches(newPodLabels) {
+				if podSelector.Matches(oldPodLabels) {
+					err = oc.deletePodEgressIPAssignments(egressIP.Name, egressIP.Status.Items, oldPod)
+					if err != nil {
+						return err
+					}
+				}
+				// egressIP has a podSelector that pod does not match: skip it
+				continue
 			}
 			// If the podSelector is empty (i.e: the EgressIP object is intended
 			// to match all pods in the namespace) and the pod has been deleted:
 			// "new" will be nil and we need to remove the setup
 			if podSelector.Empty() && new == nil {
-				return oc.deletePodEgressIPAssignments(egressIP.Name, egressIP.Status.Items, oldPod)
+				err = oc.deletePodEgressIPAssignments(egressIP.Name, egressIP.Status.Items, oldPod)
+				if err != nil {
+					return err
+				}
+				continue
```

Comment:
Why is there a continue statement here? A pod can and should only match one EgressIP object. If multiple match, then that's a user error, not an OVN-Kubernetes one. 

The bug I found ([and fixed](https://github.com/ovn-org/ovn-kubernetes/pull/2734/commits/60dace3da97e58f4cd220e11e76a33ae6f4c3c5b)) does the same thing as you do above though, so that continue seems right.

Reply:
@alexanderConstantinescu :
good question. in my tests, I've noticed that when user configs overlapping egressips, like in this bug, the route policies and nats are added to both! since we do not know which of the many this iterator ordered by, we need to ensure that all potential matches are visited.

Reply:
As you saw, my PR highlights multiple bugs with the current code, I think the bug reported is hitting a variety of them. I don't however see how the user did anything wrong by overlapping any configuration. Could you highlight that if you think that is the case? All three egress IP objects created by QE are separated and do not overlap as I see it. We obviously have a problem with the pod selector which both you and I have fixed, but I think this continue statement which applies to the namespace is not an issue.

Reply:
@alexanderConstantinescu : in this particular bz, user configured 2 egressips. Both of them use the same ns label. One of the egress uses no pod selector. In such case, we cannot really predict with of the 2 egress ips a pod may end up using. Then, later on the pod is deleted and this logic takes place. Once again, we cannot predict which of the 2 egress ips would be iterated first, but if it happens to be the one that pod was not using, we will attempt to clean up on the wrong one and return. Thus, the rows remains in ovn. Hope this make sense. ;)

Reply:
@alexanderConstantinescu I rebased this PR, so it sits on master. PTAL